RealmSwift/Sync.swift

```diff
@@ -253,7 +252,35 @@ public enum ClientResetMode {
     /// ```
     ///  For more details on the second block: ((Realm, Realm) -> Void)? = nil,
     /// - see: `RLMClientResetAfterBlock`
+    @available(*, deprecated, message: "Use discardUnsyncedChanges")
     case discardLocal(((Realm) -> Void)? = nil, ((Realm, Realm) -> Void)? = nil)
+    /// Identical in behavior to `.discardLocal.` Will fully replace
+    /// `.discardLocal.` once it's removed.
+    case discardUnsyncedChanges(((Realm) -> Void)? = nil, ((Realm, Realm) -> Void)? = nil)
```

Comment:
I supposed this callback are beforeClientReset and afterClientReset, but are not fully documented here and probably needs an argument name

Reply:
I think argument labels might be a good idea here. `case discardUnsyncedChanges(beforeReset: ((Realm) -> Void)? = nil, afterReset: ((Realm, Realm) -> Void)? = nil)` still isn't super clear but it's a bit more obvious?

Might also be better to spell out Synchronized? We do abbreviate it a lot, but using "synced" in a sentence is always kinda weird.

Reply:
I don't remember why, but there was a point in time I didn't think a typealias would work here. I was wrong. What about this?

public typealias beforeBlock = (_ beforeRealm: Realm) -> Void
public typealias afterBlock = (_ beforeRealm: Realm, _ afterRealm: Realm) -> Void
...
then
...
case discardUnsyncedChanges(beforeBlock? = nil, afterBlock? = nil)

That way I can pull out all the swift resetblock docs to one place, the typealias docstrings.

Reply:
Still need the underscore because function types cannot have argument labels.

Reply:
alternative names:

public typealias beforeResetBlock = (_ beforeResetRealm: Realm) -> Void
public typealias afterResetBlock = (_ beforeResetRealm: Realm, _ afterResetRealm: Realm) -> Void

Reply:
I'm going with 
case recoverUnsyncedChanges(beforeResetBlock? = nil, afterResetBlock? = nil)

Reply:
No "Block". Argument/tuple labels in Swift shouldn't repeat the type of the thing they're labeling (unlike in objc).

Reply:
Okay if I understand correctly I'll go with:
public typealias beforeReset = (_ beforeReset: Realm) -> Void
public typealias afterReset = (_ beforeReset: Realm, _ afterResetRealm: Realm) -> Void

case recoverUnsyncedChanges(beforeReset? = nil, afterReset? = nil)

Reply:
Oh, sorry, I don't mean that there should be typealiases here; there should be labels for the tuple elements. The problem is that at the place where the enum is being instantiated it's not obvious what the two functions are to someone reading it who isn't super familiar with the API, and having it be `. discardUnsyncedChanges(beforeReset: { realm in ...}, afterReset: { oldRealm, newRealm in ...})` would hopefully help with that.